WordPressVIPMinimum/Tests/Compatibility/RestrictedCacheGroupUnitTest.inc

```diff
@@ -0,0 +1,29 @@
+<?php
+
+$data = [ 'foo', 'bar' ];
+
+// Error
+wp_cache_set( 'cache_key', $data, 'users', 10 * MINUTE_IN_SECONDS );
+wp_cache_set( 'cache_key', $data, "userlogins" );
+wp_cache_add( $cache_key, 'data', 'posts' );
+wp_cache_add( $cache_key,   $data,   "category_relationships", 600000 );
+wp_cache_add(
+	$key,
+	$data,
+	'usermeta'
+);
+wp_cache_set( 'key',
+// Random comment to throw things off but notrly.
+
+[ 1, 2, 3 ], 'userslugs'
+);
+wp_cache_set( $key, [ 1, 2, 3 ],        'post_meta' );
+wp_cache_add($cache, "1", 'themes');
+
+// OK
+$key = 'cache_key';
+wp_cache_set( $key, $data, 'users_stuff' );
+wp_cache_add( $key, 'data', "posts_" );
+wp_cache_set( $key, $data, 'custom_taxonomy_relationships' );
+wp_cache_set( 'key', 'data', $userslugs );
+wp_cache_add( $key, 'post_meta', 'group' );
```

Comment:
What about adding some test cases with partially qualified/fully qualified function and method calls ?
This should make sure that if the parent sniff is changed (when moving to PHPCSUtils), we safeguard that that doesn't break anything.

```php
\wp_cache_set( 'cache_key', $data, "userlogins" ); // Error.
namespace\wp_cache_set( 'cache_key', $data, "userlogins" ); // Ignore.
MyNamespace\wp_cache_set( 'cache_key', $data, "userlogins" ); // Ignore.
$myOwnObj->wp_cache_set( 'cache_key', $data, "userlogins" ); // Ignore.
ClassName::wp_cache_set( 'cache_key', $data, "userlogins" ); // Ignore.
```


Nitpick: All files should end on a new line character.